source/manuals/programming-languages/nodejs/index.html.md.erb

```diff
@@ -82,24 +120,343 @@ Keeping tests in the same directory makes them easier to find and more obvious w
 |   └── product.njk
 ```
 
-#### Use a config directory
-Keep your configuration together in a config folder.
 
+See also:
+
+- [Node.js Project Structure Tutorial](https://blog.risingstack.com/node-hero-node-js-project-structure-tutorial/) on [Rising Stack](https://blog.risingstack.com/)
+
+## Language constructs
+
+### Declarations
+
+> Use `const` and `let`, avoid `var`
+
+Embrace immutability (more below) and don't let
+[hoisting](http://www.adequatelygood.com/JavaScript-Scoping-and-Hoisting.html#declarations_names_and_hoisting)
+confuse you.
+
+
+### Functions
+
+> Prefer `function` for top-level function declarations and `=>` for function literals
+
+```js
+// Use:
+const foo = function (x) {
```

Comment:
What is the reason for preferring this form over a named function `function foo(x) { ... }`?

Reply:
It's to avoid any confusion that would be caused by hoisting.

Reply:
Ok, that's what I thought 👍 . Is it worth making a note about it either in this section or in the previous one?

Reply:
Ah apologies, I just saw the comment about not using this PR for further comments.